bucket-cache.go

```diff
@@ -125,7 +125,7 @@ func processBucketLocationResponse(resp *http.Response, bucketName string) (buck
 			// For access denied error, it could be an anonymous
 			// request. Move forward and let the top level callers
 			// succeed if possible based on their policy.
-			if errResp.Code == "AccessDenied" && strings.Contains(errResp.Message, "Access Denied") {
+			if errResp.Code == "AccessDenied" && (errResp.Message == "" || strings.Contains(errResp.Message, "Access Denied")) {
```

Comment:
@harshavardhana I would have no problem with removing the check on `errResp.message` entirely, I just was not aware of why that check existed and did not want to risk a regression against another system. 

The conditions date back to when this code was first introduced in `api-error-response.go` in commit a856d4312629f7edd58dcd0cbaa973b1e7672493 but there does not appear to be any note there as to why the `Message` was a part of the condition. As you were the committer, if you have reason to think that `errResp.code == "AccessDenied"` is sufficient, I'll change to that.